### DISCUSSION TITLE
20240403 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#88] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - 
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@mickvandijke) - 
- Iván Santana(@ivan9989-pkj) - x
- Ismael Torrado(@Lepisma07) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Website

PR Review:

- https://github.com/torrust/torrust-website/pull/159

Issues:

- [x] https://github.com/torrust/torrust-website/issues/154

Other tasks:

@cgbosse and I are recreating the website structure and contents on Miro:

- Adding new sections
- New texts
- ...

### Torrust Index GUI

PR Review:

- https://github.com/torrust/torrust-index-gui/pull/510
- https://github.com/torrust/torrust-index-gui/pull/416

### Torrust Index

PR Review:

- https://github.com/torrust/torrust-index/pull/520
- https://github.com/torrust/torrust-index/pull/552

### Torrust Tracker

PR Review:

- https://github.com/torrust/torrust-tracker/pull/690
- https://github.com/torrust/torrust-tracker/pull/765

Contributor Issue Review and Resolution:

- [ ] https://github.com/torrust/torrust-tracker/issues/762. One community member asking about how to integrate the tracker with a PHP Index. 
- [ ] https://github.com/torrust/torrust-tracker/issues/771. In order to allow users to migrate to the Torrust Tracker, we need to implement a new feature to submit preexisting tracker keys. I've opened a new [epic](https://github.com/torrust/torrust-tracker/issues/773).
- [ ] https://github.com/torrust/torrust-tracker/issues/772. Another community member asking for aggregate peer statistics. There is a discussion on the issue about how to implement it. cc @da2ce7.


Issues:

- [x] https://github.com/torrust/torrust-tracker/issues/746. Use [valgrind](https://valgrind.org/) and [kcachegrind](https://kcachegrind.github.io/) to do profiling of the UDP tracker.
- [x] https://github.com/torrust/torrust-tracker/issues/768

PR:

- https://github.com/torrust/torrust-tracker/pull/751. This was a test. I wanted to know if this small refactor would improve the tracker's performance. The main idea was to merge the update and get statistics operations in the same lock.

New Issues:

- [ ] https://github.com/torrust/torrust-tracker/issues/773

New Discussions:

-  https://github.com/torrust/torrust-tracker/discussions/763. One of the community members wants to use the Tracker from a PHP application. Having a PHP wrapper for the Tracker API would make integration easier. That's not the only problem, [the other problem is we don't allow reusing pre-existing keys](https://github.com/torrust/torrust-tracker/issues/771).
- https://github.com/torrust/torrust-tracker/discussions/774. I've opened a discussion to collect all the information related to the tracker performance. I have explained the current implementation and written down all the techniques I've seen in other Rust trackers ([torrut-actixweb](https://github.com/Power2All/torrust-actix) and [aquatic](https://github.com/greatest-ape/aquatic)).
- https://github.com/torrust/torrust-tracker/discussions/775. I've opened another discussion about the problem with the API endpoint returning all torrents. This problem is related to the performance discussion.

Research:

For the [discussion about performace](https://github.com/torrust/torrust-tracker/discussions/774) I was taking a look at other implementations.

Learning Rust:

- I still consider myself a newbie in Rust. There are some parts in the code I don't fully understand. Especially concurrency and smart pointers. In fact, I would like @da2ce7 and @mickvandijke to review what I wrote in the [performance improvements discussion](https://github.com/torrust/torrust-tracker/discussions/774). I'm not sure that all I've written is 100% exact. That's the reason I have started reading another Rust book. I have almost finished reading it. THe book is:

https://nostarch.com/rust-programming-language-2nd-edition

I've found it easier to follow than the other one I read:

https://www.oreilly.com/library/view/programming-rust-2nd/9781492052586/

But maybe it's just because now I have more experience with Rust.

### Aquatic Bencher

I've added the Torrust Tracker to the aquatic Bencher. It's a tool to compare UDP tracker's performance.

PR:

- https://github.com/greatest-ape/aquatic/pull/191

**Comment by grmbyrn** [comment]
## Torrust Website
* [Fix cover image stretching across page, add blog posts to each contributor's page](https://github.com/torrust/torrust-website/pull/159)

## SvelteKit Practice
* [Frontend Mentor](https://github.com/grmbyrn/coffeeroasters-sk)
* Learned Prisma for use with auth

## Rust
* Began learning Rust through [YouTube course](https://www.youtube.com/watch?v=lzKeecy4OmQ&ab_channel=ZeroToMastery)

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/issues/401
- https://github.com/torrust/torrust-index-gui/pull/510

### Index
	
- https://github.com/torrust/torrust-index/pull/552
- https://github.com/torrust/torrust-index/issues/448

### Training/Research


- Tidy first? book
